--- a/configure	2012-09-22 15:07:41 +0400
+++ b/configure	2012-09-22 15:44:40 +0400
@@ -7955,23 +7955,23 @@
 
 
 # See if the user wants to add (or not) GNU Guile support
-PKG_PROG_PKG_CONFIG
-
+#PKG_PROG_PKG_CONFIG
+#
 # Check whether --with-guile was given.
-if test "${with_guile+set}" = set; then :
-  withval=$with_guile;
-fi
-
-
+#if test "${with_guile+set}" = set; then :
+#  withval=$with_guile;
+#fi
+#
+#
 # For some strange reason, at least on Ubuntu, each version of Guile
 # comes with it's own PC file so we have to specify them as individual
 # packages.  Ugh.
-if test "x$with_guile" != xno; then :
-   PKG_CHECK_MODULES(GUILE, guile-2.0, have_guile=yes,
-  PKG_CHECK_MODULES([GUILE], [guile-1.8], [have_guile=yes],
-    [have_guile=no]))
-
-fi
+#if test "x$with_guile" != xno; then :
+#   PKG_CHECK_MODULES(GUILE, guile-2.0, have_guile=yes,
+#  PKG_CHECK_MODULES([GUILE], [guile-1.8], [have_guile=yes],
+#    [have_guile=no]))
+#
+#fi
 
 if test "$have_guile" = yes; then :
 
