--- gcc/config/i386/i386.c	2015-11-18 16:45:26.000000000 +0100
+++ gcc/config/i386/i386.c	mingwpy working copy
@@ -4331,14 +4331,8 @@ ix86_option_override_internal (bool main
   else if (opts_set->x_target_flags & MASK_RECIP)
     opts->x_recip_mask &= ~(RECIP_MASK_ALL & ~opts->x_recip_mask_explicit);
 
-  /* Default long double to 64-bit for 32-bit Bionic and to __float128
-     for 64-bit Bionic.  */
-  if (TARGET_HAS_BIONIC
-      && !(opts_set->x_target_flags
-	   & (MASK_LONG_DOUBLE_64 | MASK_LONG_DOUBLE_128)))
-    opts->x_target_flags |= (TARGET_64BIT
-			     ? MASK_LONG_DOUBLE_128
-			     : MASK_LONG_DOUBLE_64);
+  /* Default long double to 64-bit for mingwpy builds */
+   if !(opts_set->x_target_flags & (MASK_LONG_DOUBLE_64 | MASK_LONG_DOUBLE_128))
+    opts->x_target_flags |= MASK_LONG_DOUBLE_64;
 
   /* Only one of them can be active.  */
   gcc_assert ((opts->x_target_flags & MASK_LONG_DOUBLE_64) == 0
