--- a/gcc/gcc.c
+++ b/gcc/gcc.c
@@ -10201,7 +10201,7 @@
   int i;
   /* Must have exactly two arguments.  */
   if (argc != 2)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   for (i = 0; i < n_infiles; i++)
     {
@@ -10222,7 +10222,7 @@
   int i;
   /* Must have exactly one argument.  */
   if (argc != 1)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   for (i = 0; i < n_infiles; i++)
     {
@@ -10244,17 +10244,17 @@
 
   if (regcomp (&r, "^([1-9][0-9]*|0)(\\.([1-9][0-9]*|0))*$",
 	       REG_EXTENDED | REG_NOSUB) != 0)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
   rresult = regexec (&r, v1, 0, NULL, 0);
   if (rresult == REG_NOMATCH)
     fatal_error (input_location, "invalid version number %qs", v1);
   else if (rresult != 0)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
   rresult = regexec (&r, v2, 0, NULL, 0);
   if (rresult == REG_NOMATCH)
     fatal_error (input_location, "invalid version number %qs", v2);
   else if (rresult != 0)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   return strverscmp (v1, v2);
 }
@@ -10297,7 +10297,7 @@
   if (argc < 3)
     fatal_error (input_location, "too few arguments to %%:version-compare");
   if (argv[0][0] == '\0')
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
   if ((argv[0][1] == '<' || argv[0][1] == '>') && argv[0][0] != '!')
     nargs = 2;
   if (argc != nargs + 3)
@@ -10362,7 +10362,7 @@
   char *file;
 
   if (argc != 1)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   file = find_a_file (&startfile_prefixes, argv[0], R_OK, true);
   read_specs (file ? file : argv[0], false, false);
@@ -10379,7 +10379,7 @@
   const char *file;
 
   if (argc != 1)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   file = find_file (argv[0]);
   return file;
@@ -10395,7 +10395,7 @@
   const char *option;
 
   if (argc != 0)
-    abort ();
+    fancy_abort (__FILE__, __LINE__, __FUNCTION__);
 
   option = concat ("-iplugindir=", find_file ("plugin"), NULL);
   return option;
--- a/gcc/system.h
+++ b/gcc/system.h
@@ -772,7 +772,6 @@
    reporting the location of the error in the source file.  */
 extern void fancy_abort (const char *, int, const char *)
 					 ATTRIBUTE_NORETURN ATTRIBUTE_COLD;
-#define abort() fancy_abort (__FILE__, __LINE__, __FUNCTION__)
 
 /* Use gcc_assert(EXPR) to test invariants.  */
 #if ENABLE_ASSERT_CHECKING
--- a/libcpp/system.h
+++ b/libcpp/system.h
@@ -401,7 +401,6 @@
 /* Redefine abort to report an internal error w/o coredump, and
    reporting the location of the error in the source file.  */
 extern void fancy_abort (const char *, int, const char *) ATTRIBUTE_NORETURN;
-#define abort() fancy_abort (__FILE__, __LINE__, __FUNCTION__)
 
 /* Use gcc_assert(EXPR) to test invariants.  */
 #if ENABLE_ASSERT_CHECKING
