--- a/gcc/stmt.c.orig	2012-11-06 19:40:39 +0400
+++ b/gcc/stmt.c	2012-11-06 19:40:57 +0400
@@ -2061,7 +2061,7 @@
       tree lab = CASE_LABEL (elt);
       basic_block case_bb = label_to_block_fn (cfun, lab);
       edge case_edge = find_edge (bb, case_bb);
-      case_edge->aux = (void *)((long)(case_edge->aux) + 1);
+      case_edge->aux = (void *)((intptr_t)(case_edge->aux) + 1);
     }
 }
 
@@ -2176,7 +2176,7 @@
       edge case_edge = find_edge (bb, case_bb);
       case_list = add_case_node (
           case_list, low, high, lab,
-          case_edge->probability / (long)(case_edge->aux),
+          case_edge->probability / (intptr_t)(case_edge->aux),
           case_node_pool);
     }
   pointer_set_destroy (seen_labels);
