--- gcc-5.1.0/config/override.m4.orig    2015-05-01 16:38:03.348243300 -0400
+++ gcc-5.1.0/config/override.m4 2015-05-02 00:30:58.345729900 -0400
@@ -29,7 +29,7 @@
 
 dnl Ensure exactly this Autoconf version is used
 m4_ifndef([_GCC_AUTOCONF_VERSION],
-  [m4_define([_GCC_AUTOCONF_VERSION], [2.64])])
+  [m4_define([_GCC_AUTOCONF_VERSION], [2.69])])
 
 dnl Test for the exact version when AC_INIT is expanded.
 dnl This allows to update the tree in steps (for testing)
