Index: libatomic/configure.tgt
===================================================================
--- libatomic/configure.tgt	(revision 227599)
+++ libatomic/configure.tgt	(working copy)
@@ -121,7 +121,7 @@
 	# OS support for atomic primitives.
         case ${target_thread_file} in
           win32)
-            config_path="${config_path} mingw"
+            config_path="${config_path} posix"
             ;;
           posix)
             config_path="${config_path} posix"
