diff -rupN gcc-5.3.0/gcc/config/i386/i386.c gcc-5.3.0_2/gcc/config/i386/i386.c
--- gcc-5.3.0/gcc/config/i386/i386.c	2015-11-18 16:45:26.000000000 +0100
+++ gcc-5.3.0_2/gcc/config/i386/i386.c	2016-02-22 22:48:07.134823300 +0100
@@ -4339,6 +4339,10 @@ ix86_option_override_internal (bool main
     opts->x_target_flags |= (TARGET_64BIT
 			     ? MASK_LONG_DOUBLE_128
 			     : MASK_LONG_DOUBLE_64);
+  
+  /* Default long double to 64-bit for mingwpy.  */
+   if (!(opts_set->x_target_flags & (MASK_LONG_DOUBLE_64 | MASK_LONG_DOUBLE_128)))
+	opts->x_target_flags |= MASK_LONG_DOUBLE_64;
 
   /* Only one of them can be active.  */
   gcc_assert ((opts->x_target_flags & MASK_LONG_DOUBLE_64) == 0
