diff -urN gcc-4.7.0_orig/gcc/config/i386/mingw32.h gcc-4.7.0/gcc/config/i386/mingw32.h
--- gcc-4.7.0_orig/gcc/config/i386/mingw32.h	2012-02-13 23:46:38.000000000 +0200
+++ gcc-4.7.0/gcc/config/i386/mingw32.h	2017-10-13 21:46:02.258819600 +0300
@@ -150,6 +150,11 @@
 #define STANDARD_STARTFILE_PREFIX_2 ""
 #endif
 
+/* For native mingw-version we need to take care that NATIVE_SYSTEM_HEADER_DIR
+   macro contains POSIX-style path.  See bug 52947.  */
+#undef NATIVE_SYSTEM_HEADER_DIR
+#define NATIVE_SYSTEM_HEADER_DIR "/mingw/include"
+
 /* Output STRING, a string representing a filename, to FILE.
    We canonicalize it to be in Unix format (backslashes are replaced
    forward slashes.  */
