diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-04-24 00:23:24.486300285 +0100
+++ b/setup.py	2013-04-24 00:23:24.874303338 +0100
@@ -1323,7 +1323,7 @@
 
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
-        if host_platform not in ['cygwin']:
+        if host_platform not in ['cygwin', 'mingw', 'win32']:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg for arg in config_args
