diff --git a/mingw-w64-tools/gendef/src/fsredir.c b/mingw-w64-tools/gendef/src/fsredir.c
index 36afd838..ee924feb 100644
--- a/mingw-w64-tools/gendef/src/fsredir.c
+++ b/mingw-w64-tools/gendef/src/fsredir.c
@@ -25,8 +25,8 @@
 
 static PVOID revert; /*revert pointer*/
 static HMODULE kernel32handle;
-typedef WINBOOL (__stdcall (*redirector))(PVOID *);
-typedef WINBOOL (__stdcall (*revertor))(PVOID);
+typedef size_t (__stdcall (*redirector))(PVOID *);
+typedef size_t (__stdcall (*revertor))(PVOID);
 static redirector redirectorfunction; /*Wow64DisableWow64FsRedirection*/
 static revertor revertorfunction;     /*Wow64RevertWow64FsRedirection*/

