--- a/configure.orig	2010-11-15 02:48:06 +0400
+++ b/configure	2012-09-30 23:01:37 +0400
@@ -8181,13 +8181,13 @@
 if test "x$ac_cv_lib_curses_tgetent" = x""yes; then
   bash_cv_termcap_lib=libcurses
 else
-  { $as_echo "$as_me:$LINENO: checking for tgetent in -lncurses" >&5
-$as_echo_n "checking for tgetent in -lncurses... " >&6; }
+  { $as_echo "$as_me:$LINENO: checking for tgetent in -lncursesw" >&5
+$as_echo_n "checking for tgetent in -lncursesw... " >&6; }
 if test "${ac_cv_lib_ncurses_tgetent+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lncurses  $LIBS"
+LIBS="-lncursesw  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -8247,7 +8247,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_ncurses_tgetent" >&5
 $as_echo "$ac_cv_lib_ncurses_tgetent" >&6; }
 if test "x$ac_cv_lib_ncurses_tgetent" = x""yes; then
-  bash_cv_termcap_lib=libncurses
+  bash_cv_termcap_lib=libncursesw
 else
   bash_cv_termcap_lib=gnutermcap
 fi
@@ -8278,8 +8278,8 @@
 elif test $bash_cv_termcap_lib = libtinfo; then
 TERMCAP_LIB=-ltinfo
 TERMCAP_DEP=
-elif test $bash_cv_termcap_lib = libncurses; then
-TERMCAP_LIB=-lncurses
+elif test $bash_cv_termcap_lib = libncursesw; then
+TERMCAP_LIB=-lncursesw
 TERMCAP_DEP=
 elif test $bash_cv_termcap_lib = libc; then
 TERMCAP_LIB=
