diff -urN tcl8.6.9_orig/pkgs/itcl4.1.2/tclconfig/tcl.m4 tcl8.6.9/pkgs/itcl4.1.2/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/itcl4.1.2/tclconfig/tcl.m4	2018-11-01 17:59:24.000000000 +0200
+++ tcl8.6.9/pkgs/itcl4.1.2/tclconfig/tcl.m4	2018-12-10 08:30:46.610518700 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/sqlite3.25.3/configure tcl8.6.9/pkgs/sqlite3.25.3/configure
--- tcl8.6.9_orig/pkgs/sqlite3.25.3/configure	2018-11-12 20:03:27.000000000 +0200
+++ tcl8.6.9/pkgs/sqlite3.25.3/configure	2018-12-10 08:30:46.610518700 +0200
@@ -8455,9 +8455,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/sqlite3.25.3/tclconfig/tcl.m4 tcl8.6.9/pkgs/sqlite3.25.3/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/sqlite3.25.3/tclconfig/tcl.m4	2018-11-16 21:01:50.000000000 +0200
+++ tcl8.6.9/pkgs/sqlite3.25.3/tclconfig/tcl.m4	2018-12-10 08:30:46.610518700 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbc1.1.0/tclconfig/tcl.m4 tcl8.6.9/pkgs/tdbc1.1.0/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/tdbc1.1.0/tclconfig/tcl.m4	2018-11-01 18:02:25.000000000 +0200
+++ tcl8.6.9/pkgs/tdbc1.1.0/tclconfig/tcl.m4	2018-12-10 08:30:46.610518700 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbcmysql1.1.0/tclconfig/tcl.m4 tcl8.6.9/pkgs/tdbcmysql1.1.0/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/tdbcmysql1.1.0/tclconfig/tcl.m4	2018-11-01 18:08:20.000000000 +0200
+++ tcl8.6.9/pkgs/tdbcmysql1.1.0/tclconfig/tcl.m4	2018-12-10 08:30:46.610518700 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbcodbc1.1.0/tclconfig/tcl.m4 tcl8.6.9/pkgs/tdbcodbc1.1.0/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/tdbcodbc1.1.0/tclconfig/tcl.m4	2018-11-01 18:09:16.000000000 +0200
+++ tcl8.6.9/pkgs/tdbcodbc1.1.0/tclconfig/tcl.m4	2018-12-10 08:30:46.626141400 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbcpostgres1.1.0/configure tcl8.6.9/pkgs/tdbcpostgres1.1.0/configure
--- tcl8.6.9_orig/pkgs/tdbcpostgres1.1.0/configure	2018-11-01 18:09:49.000000000 +0200
+++ tcl8.6.9/pkgs/tdbcpostgres1.1.0/configure	2018-12-10 08:30:46.626141400 +0200
@@ -8910,9 +8910,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbcpostgres1.1.0/tclconfig/tcl.m4 tcl8.6.9/pkgs/tdbcpostgres1.1.0/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/tdbcpostgres1.1.0/tclconfig/tcl.m4	2018-11-01 18:09:45.000000000 +0200
+++ tcl8.6.9/pkgs/tdbcpostgres1.1.0/tclconfig/tcl.m4	2018-12-10 08:30:46.626141400 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/tdbcsqlite3-1.1.0/tclconfig/tcl.m4 tcl8.6.9/pkgs/tdbcsqlite3-1.1.0/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/tdbcsqlite3-1.1.0/tclconfig/tcl.m4	2018-11-01 18:10:23.000000000 +0200
+++ tcl8.6.9/pkgs/tdbcsqlite3-1.1.0/tclconfig/tcl.m4	2018-12-10 08:30:46.626141400 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/pkgs/thread2.8.4/tclconfig/tcl.m4 tcl8.6.9/pkgs/thread2.8.4/tclconfig/tcl.m4
--- tcl8.6.9_orig/pkgs/thread2.8.4/tclconfig/tcl.m4	2018-11-01 18:11:01.000000000 +0200
+++ tcl8.6.9/pkgs/thread2.8.4/tclconfig/tcl.m4	2018-12-10 08:30:46.626141400 +0200
@@ -3222,9 +3222,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.9_orig/win/tcl.m4 tcl8.6.9/win/tcl.m4
--- tcl8.6.9_orig/win/tcl.m4	2018-12-10 08:21:40.537254700 +0200
+++ tcl8.6.9/win/tcl.m4	2018-12-10 08:30:46.626141400 +0200
@@ -635,7 +635,6 @@
 
     if test "${GCC}" = "yes" ; then
 	extra_cflags="-pipe"
-	extra_ldflags="-pipe -static-libgcc"
 	AC_CACHE_CHECK(for mingw32 version of gcc,
 	    ac_cv_win32,
 	    AC_TRY_COMPILE([
@@ -691,6 +690,7 @@
 	if test "${SHARED_BUILD}" = "0" ; then
 	    # static
             AC_MSG_RESULT([using static flags])
+            extra_ldflags="-pipe -static-libgcc"
 	    runtime=
 	    LIBRARIES="\${STATIC_LIBRARIES}"
 	    EXESUFFIX="s\${DBGX}.exe"
