diff -urN tcl8.6.7_orig/pkgs/itcl4.1.0/tclconfig/tcl.m4 tcl8.6.7/pkgs/itcl4.1.0/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/itcl4.1.0/tclconfig/tcl.m4	2017-08-09 17:45:56.000000000 +0300
+++ tcl8.6.7/pkgs/itcl4.1.0/tclconfig/tcl.m4	2017-10-11 19:53:38.278631900 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/sqlite3.20.0/configure tcl8.6.7/pkgs/sqlite3.20.0/configure
--- tcl8.6.7_orig/pkgs/sqlite3.20.0/configure	2017-08-04 14:57:41.000000000 +0300
+++ tcl8.6.7/pkgs/sqlite3.20.0/configure	2017-10-11 19:53:38.294258900 +0300
@@ -8890,9 +8890,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/sqlite3.20.0/tclconfig/tcl.m4 tcl8.6.7/pkgs/sqlite3.20.0/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/sqlite3.20.0/tclconfig/tcl.m4	2017-08-09 17:45:57.000000000 +0300
+++ tcl8.6.7/pkgs/sqlite3.20.0/tclconfig/tcl.m4	2017-10-11 19:53:38.294258900 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbc1.0.5/tclconfig/tcl.m4 tcl8.6.7/pkgs/tdbc1.0.5/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/tdbc1.0.5/tclconfig/tcl.m4	2017-04-18 19:49:49.000000000 +0300
+++ tcl8.6.7/pkgs/tdbc1.0.5/tclconfig/tcl.m4	2017-10-11 19:53:38.294258900 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbcmysql1.0.5/tclconfig/tcl.m4 tcl8.6.7/pkgs/tdbcmysql1.0.5/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/tdbcmysql1.0.5/tclconfig/tcl.m4	2017-04-18 19:50:46.000000000 +0300
+++ tcl8.6.7/pkgs/tdbcmysql1.0.5/tclconfig/tcl.m4	2017-10-11 19:53:38.309882500 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbcodbc1.0.5/tclconfig/tcl.m4 tcl8.6.7/pkgs/tdbcodbc1.0.5/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/tdbcodbc1.0.5/tclconfig/tcl.m4	2017-08-09 17:45:57.000000000 +0300
+++ tcl8.6.7/pkgs/tdbcodbc1.0.5/tclconfig/tcl.m4	2017-10-11 19:53:38.309882500 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbcpostgres1.0.5/configure tcl8.6.7/pkgs/tdbcpostgres1.0.5/configure
--- tcl8.6.7_orig/pkgs/tdbcpostgres1.0.5/configure	2017-08-09 17:45:57.000000000 +0300
+++ tcl8.6.7/pkgs/tdbcpostgres1.0.5/configure	2017-10-11 19:53:38.309882500 +0300
@@ -9429,9 +9429,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbcpostgres1.0.5/tclconfig/tcl.m4 tcl8.6.7/pkgs/tdbcpostgres1.0.5/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/tdbcpostgres1.0.5/tclconfig/tcl.m4	2017-08-09 17:45:57.000000000 +0300
+++ tcl8.6.7/pkgs/tdbcpostgres1.0.5/tclconfig/tcl.m4	2017-10-11 19:53:38.325508300 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/tdbcsqlite3-1.0.5/tclconfig/tcl.m4 tcl8.6.7/pkgs/tdbcsqlite3-1.0.5/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/tdbcsqlite3-1.0.5/tclconfig/tcl.m4	2017-08-09 17:45:58.000000000 +0300
+++ tcl8.6.7/pkgs/tdbcsqlite3-1.0.5/tclconfig/tcl.m4	2017-10-11 19:53:38.325508300 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/pkgs/thread2.8.1/tclconfig/tcl.m4 tcl8.6.7/pkgs/thread2.8.1/tclconfig/tcl.m4
--- tcl8.6.7_orig/pkgs/thread2.8.1/tclconfig/tcl.m4	2017-08-09 17:45:58.000000000 +0300
+++ tcl8.6.7/pkgs/thread2.8.1/tclconfig/tcl.m4	2017-10-11 19:53:38.325508300 +0300
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.7_orig/win/tcl.m4 tcl8.6.7/win/tcl.m4
--- tcl8.6.7_orig/win/tcl.m4	2017-10-11 19:43:44.540807800 +0300
+++ tcl8.6.7/win/tcl.m4	2017-10-11 19:53:38.325508300 +0300
@@ -635,7 +635,6 @@
 
     if test "${GCC}" = "yes" ; then
 	extra_cflags="-pipe"
-	extra_ldflags="-pipe -static-libgcc"
 	AC_CACHE_CHECK(for mingw32 version of gcc,
 	    ac_cv_win32,
 	    AC_TRY_COMPILE([
@@ -691,6 +690,7 @@
 	if test "${SHARED_BUILD}" = "0" ; then
 	    # static
             AC_MSG_RESULT([using static flags])
+            extra_ldflags="-pipe -static-libgcc"
 	    runtime=
 	    LIBRARIES="\${STATIC_LIBRARIES}"
 	    EXESUFFIX="s\${DBGX}.exe"
