diff -urN tcl8.6.8_orig/pkgs/itcl4.1.1/tclconfig/tcl.m4 tcl8.6.8/pkgs/itcl4.1.1/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/itcl4.1.1/tclconfig/tcl.m4	2017-12-21 20:16:24.000000000 +0200
+++ tcl8.6.8/pkgs/itcl4.1.1/tclconfig/tcl.m4	2018-02-14 14:51:37.701603400 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/sqlite3.21.0/configure tcl8.6.8/pkgs/sqlite3.21.0/configure
--- tcl8.6.8_orig/pkgs/sqlite3.21.0/configure	2017-12-21 15:58:19.000000000 +0200
+++ tcl8.6.8/pkgs/sqlite3.21.0/configure	2018-02-14 14:51:37.717214700 +0200
@@ -8880,9 +8880,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/sqlite3.21.0/tclconfig/tcl.m4 tcl8.6.8/pkgs/sqlite3.21.0/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/sqlite3.21.0/tclconfig/tcl.m4	2017-12-21 20:16:24.000000000 +0200
+++ tcl8.6.8/pkgs/sqlite3.21.0/tclconfig/tcl.m4	2018-02-14 14:51:37.717214700 +0200
@@ -3427,9 +3427,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbc1.0.6/tclconfig/tcl.m4 tcl8.6.8/pkgs/tdbc1.0.6/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/tdbc1.0.6/tclconfig/tcl.m4	2017-12-14 15:33:28.000000000 +0200
+++ tcl8.6.8/pkgs/tdbc1.0.6/tclconfig/tcl.m4	2018-02-14 14:51:37.717214700 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbcmysql1.0.6/tclconfig/tcl.m4 tcl8.6.8/pkgs/tdbcmysql1.0.6/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/tdbcmysql1.0.6/tclconfig/tcl.m4	2017-12-14 15:33:48.000000000 +0200
+++ tcl8.6.8/pkgs/tdbcmysql1.0.6/tclconfig/tcl.m4	2018-02-14 14:51:37.717214700 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbcodbc1.0.6/tclconfig/tcl.m4 tcl8.6.8/pkgs/tdbcodbc1.0.6/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/tdbcodbc1.0.6/tclconfig/tcl.m4	2017-12-21 20:16:24.000000000 +0200
+++ tcl8.6.8/pkgs/tdbcodbc1.0.6/tclconfig/tcl.m4	2018-02-14 14:51:37.717214700 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbcpostgres1.0.6/configure tcl8.6.8/pkgs/tdbcpostgres1.0.6/configure
--- tcl8.6.8_orig/pkgs/tdbcpostgres1.0.6/configure	2017-12-21 20:16:25.000000000 +0200
+++ tcl8.6.8/pkgs/tdbcpostgres1.0.6/configure	2018-02-14 14:51:37.717214700 +0200
@@ -9418,9 +9418,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbcpostgres1.0.6/tclconfig/tcl.m4 tcl8.6.8/pkgs/tdbcpostgres1.0.6/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/tdbcpostgres1.0.6/tclconfig/tcl.m4	2017-12-21 20:16:25.000000000 +0200
+++ tcl8.6.8/pkgs/tdbcpostgres1.0.6/tclconfig/tcl.m4	2018-02-14 14:51:37.732840400 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/tdbcsqlite3-1.0.6/tclconfig/tcl.m4 tcl8.6.8/pkgs/tdbcsqlite3-1.0.6/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/tdbcsqlite3-1.0.6/tclconfig/tcl.m4	2017-12-21 20:16:25.000000000 +0200
+++ tcl8.6.8/pkgs/tdbcsqlite3-1.0.6/tclconfig/tcl.m4	2018-02-14 14:51:37.732840400 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/pkgs/thread2.8.2/tclconfig/tcl.m4 tcl8.6.8/pkgs/thread2.8.2/tclconfig/tcl.m4
--- tcl8.6.8_orig/pkgs/thread2.8.2/tclconfig/tcl.m4	2017-12-21 20:16:25.000000000 +0200
+++ tcl8.6.8/pkgs/thread2.8.2/tclconfig/tcl.m4	2018-02-14 14:51:37.732840400 +0200
@@ -3413,9 +3413,6 @@
 		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TK_BIN_DIR}/${TK_STUB_LIB_FILE}`\""
 	    fi
 	    SHLIB_LD_LIBS="${SHLIB_LD_LIBS} \"`${CYGPATH} ${TCL_BIN_DIR}/${TCL_STUB_LIB_FILE}`\""
-	    if test "$GCC" = "yes"; then
-		SHLIB_LD_LIBS="${SHLIB_LD_LIBS} -static-libgcc"
-	    fi
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${SHARED_LIB_SUFFIX}"
 	else
 	    eval eval "PKG_LIB_FILE=${PACKAGE_LIB_PREFIX}${PACKAGE_NAME}${UNSHARED_LIB_SUFFIX}"
diff -urN tcl8.6.8_orig/win/tcl.m4 tcl8.6.8/win/tcl.m4
--- tcl8.6.8_orig/win/tcl.m4	2017-12-21 20:16:25.000000000 +0200
+++ tcl8.6.8/win/tcl.m4	2018-02-14 14:51:37.732840400 +0200
@@ -635,7 +635,6 @@
 
     if test "${GCC}" = "yes" ; then
 	extra_cflags="-pipe"
-	extra_ldflags="-pipe -static-libgcc"
 	AC_CACHE_CHECK(for mingw32 version of gcc,
 	    ac_cv_win32,
 	    AC_TRY_COMPILE([
@@ -691,6 +690,7 @@
 	if test "${SHARED_BUILD}" = "0" ; then
 	    # static
             AC_MSG_RESULT([using static flags])
+            extra_ldflags="-pipe -static-libgcc"
 	    runtime=
 	    LIBRARIES="\${STATIC_LIBRARIES}"
 	    EXESUFFIX="s\${DBGX}.exe"
