--- generic/tclLoad.c
+++ generic/tclLoad.c
@@ -795,9 +795,7 @@ Tcl_UnloadObjCmd(
 
 	    if (unLoadProcPtr != NULL) {
 		Tcl_MutexLock(&packageMutex);
-		if ((pkgPtr->unloadProc != NULL) || (unLoadProcPtr == TclFSUnloadTempFile)) {
-		    (*unLoadProcPtr)(pkgPtr->loadHandle);
-		}
+		(*unLoadProcPtr)(pkgPtr->loadHandle);
 
 		/*
 		 * Remove this library from the loaded library cache.
@@ -1154,9 +1152,7 @@ TclFinalizeLoad(void)
 
 	if (pkgPtr->fileName[0] != '\0') {
 	    Tcl_FSUnloadFileProc *unLoadProcPtr = pkgPtr->unLoadProcPtr;
-	    if ((unLoadProcPtr != NULL)
-		    && ((pkgPtr->unloadProc != NULL)
-		    || (unLoadProcPtr == TclFSUnloadTempFile))) {
+	    if (unLoadProcPtr != NULL) {
 		(*unLoadProcPtr)(pkgPtr->loadHandle);
 	    }
 	}
