--- win/Makefile.in	2009-05-23 14:03:18.000000000 +0200
+++ win/Makefile.in	2009-05-23 14:05:10.000000000 +0200
@@ -385,7 +385,7 @@
 
 winhelp: $(ROOT_DIR)/tools/man2help.tcl $(MAN2TCL)
 	TCL_LIBRARY="$(LIBRARY_DIR)"; export TCL_LIBRARY; \
-	./$(TCLSH) "$(ROOT_DIR_NATIVE)"/tools/man2help.tcl tcl "$(VER)" $(TCL_DOCS)
+	tclsh$(DOTVER) "$(ROOT_DIR_NATIVE)"/tools/man2help.tcl tcl "$(VER)" $(TCL_DOCS)
 	hcw /c /e tcl.hpj
 
 $(MAN2TCL): $(ROOT_DIR)/tools/man2tcl.c
@@ -658,13 +658,13 @@
 install-tzdata:
 	@echo "Installing time zone data"
 	@TCL_LIBRARY="${LIBRARY_DIR}"; export TCL_LIBRARY; \
-	./$(TCLSH) "$(ROOT_DIR)/tools/installData.tcl" \
+	tclsh$(DOTVER) "$(ROOT_DIR)/tools/installData.tcl" \
 	    "$(ROOT_DIR)/library/tzdata" "$(SCRIPT_INSTALL_DIR)/tzdata"
 
 install-msgs:
 	@echo "Installing message catalogs"
 	@TCL_LIBRARY="${LIBRARY_DIR}"; export TCL_LIBRARY; \
-	./$(TCLSH) "$(ROOT_DIR)/tools/installData.tcl" \
+	tclsh$(DOTVER) "$(ROOT_DIR)/tools/installData.tcl" \
 	    "$(ROOT_DIR)/library/msgs" "$(SCRIPT_INSTALL_DIR)/msgs"
 
 install-doc: doc
