--- library/init.tcl
+++ library/init.tcl
@@ -53,11 +53,6 @@ namespace eval tcl {
 	    lappend ::auto_path $Dir
 	}
     }
-    set Dir [file join [file dirname [file dirname \
-	    [info nameofexecutable]]] lib]
-    if {$Dir ni $::auto_path} {
-	lappend ::auto_path $Dir
-    }
     catch {
 	foreach Dir $::tcl_pkgPath {
 	    if {$Dir ni $::auto_path} {
--- a/library/tm.tcl
+++ b/library/tm.tcl
@@ -310,10 +310,7 @@ proc ::tcl::tm::Defaults {} {
 
     # Note that we're using [::list], not [list] because [list] means
     # something other than [::list] in this namespace.
-    roots [::list \
-	    [file dirname [info library]] \
-	    [file join [file dirname [file dirname $exe]] lib] \
-	    ]
+    roots [lreverse $::tcl_pkgPath]
 
     if {$tcl_platform(platform) eq "windows"} {
 	set sep ";"
