--- make_git/main.c.orig	2013-04-16 10:13:57 +0400
+++ make_git/main.c	2013-04-16 10:14:32 +0400
@@ -690,9 +690,9 @@
 {
   const char **pp;
 
-  if (!output_sync_option)
+#ifndef OUTPUT_SYNC
     return;
-
+#else
   for (pp=output_sync_option->list; *pp; ++pp)
     {
       const char *p = *pp;
@@ -704,6 +704,7 @@
       else
         fatal (NILF, _("unknown output-sync type '%s'"), p);
     }
+#endif
 }
 
 #ifdef WINDOWS32
