--- isl-0.12/Makefile.in.orig	2013-07-18 10:15:32.998046900 +0400
+++ isl-0.12/Makefile.in	2013-07-25 13:31:37.057617200 +0400
@@ -624,7 +624,7 @@
 
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
 libisl_la_LDFLAGS = -version-info @versioninfo@ \
-	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
+	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@ -no-undefined
 
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
 isl_test_CPPFLAGS = $(INCLUDES) @GMP_CPPFLAGS@
--- isl-0.12/Makefile.am.orig	2013-06-23 16:13:54.000000000 +0400
+++ isl-0.12/Makefile.am	2013-07-25 13:31:49.416992200 +0400
@@ -155,7 +155,7 @@
 	isl_piplib.c
 libisl_la_LIBADD = @PIPLIB_LIBS@ @GMP_LIBS@
 libisl_la_LDFLAGS = -version-info @versioninfo@ \
-	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@
+	@PIPLIB_LDFLAGS@ @GMP_LDFLAGS@ -no-undefined
 libisl_la_CPPFLAGS = $(INCLUDES) @PIPLIB_CPPFLAGS@ @GMP_CPPFLAGS@
 
 isl_test_CPPFLAGS = $(INCLUDES) @GMP_CPPFLAGS@
