diff -urN a/Makefile.pre.in b/Makefile.pre.in
--- a/Makefile.pre.in	2013-04-24 00:25:20.033164156 +0100
+++ b/Makefile.pre.in	2013-04-24 00:25:20.407166834 +0100
@@ -556,7 +556,7 @@
 # FIXME: next may fail in cross-compilation environment
 # Use a stamp file to prevent make -j invoking pgen twice
 $(GRAMMAR_H) $(GRAMMAR_C): Parser/pgen.stamp
-Parser/pgen.stamp: $(PGEN) $(GRAMMAR_INPUT)
+Parser/pgen.stamp: $(GRAMMAR_INPUT)
 		-@$(INSTALL) -d Include
 		$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 		-touch Parser/pgen.stamp
