diff -urN a-2.7.3/Makefile.pre.in b-2.7.3/Makefile.pre.in
--- a-2.7.3/Makefile.pre.in	2012-11-22 20:47:09.135061383 +0000
+++ b-2.7.3/Makefile.pre.in	2012-11-22 20:47:14.464328353 +0000
@@ -556,7 +556,7 @@
 # FIXME: next may fail in cross-compilation environment
 # Use a stamp file to prevent make -j invoking pgen twice
 $(GRAMMAR_H) $(GRAMMAR_C): Parser/pgen.stamp
-Parser/pgen.stamp: $(PGEN) $(GRAMMAR_INPUT)
+Parser/pgen.stamp: $(GRAMMAR_INPUT)
 		-@$(INSTALL) -d Include
 		$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 		-touch Parser/pgen.stamp
