diff -urN a/setup.py b/setup.py
--- a/setup.py	2013-04-24 00:25:22.122179119 +0100
+++ b/setup.py	2013-04-24 00:25:22.506181871 +0100
@@ -417,7 +417,7 @@
             # Ensure that /usr/local is always used
             add_dir_to_list(self.compiler.library_dirs, '/usr/local/lib')
             add_dir_to_list(self.compiler.include_dirs, '/usr/local/include')
-        self.add_multiarch_paths()
+            self.add_multiarch_paths()
 
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
