diff -urN a/Lib/compileall.py b/Lib/compileall.py
--- a/Lib/compileall.py	2012-04-10 00:07:29.000000000 +0100
+++ b/Lib/compileall.py	2012-10-09 14:30:45.568918880 +0100
@@ -42,6 +42,8 @@
     success = 1
     for name in names:
         fullname = os.path.join(dir, name)
+        if sys.platform == "win32" and sys.version.find("GCC") >= 0:
+            fullname = fullname.replace('\\','/')
         if ddir is not None:
             dfile = os.path.join(ddir, name)
         else:
